src/routes/CreateSafePage/steps/SelectWalletAndNetworkStep.tsx

```diff
@@ -25,7 +25,7 @@ export const selectWalletAndNetworkStepLabel = 'Connect wallet & select network'
 
 function SelectWalletAndNetworkStep(): ReactElement {
   const [isNetworkSelectorPopupOpen, setIsNetworkSelectorPopupOpen] = useState(false)
-  const isWalletConnected = !!useSelector(providerNameSelector)
+  const isWalletConnected = !!useSelector(availableSelector)
```

Comment:
'Mobile Safe' can sometimes be set in the provider when we load the QR code. It makes no sense to check whether the name exists to show the 'switch network' button, but rather that it is 'available' (loaded).